File: solutions/_965_unique_email_addresses/solution.py
from typing import List


class Solution:
    def numUniqueEmails(self, emails: list[str]) -> int:
        unique_emails: set[str] = set()
        for email in emails:
            local, domain = email.split("@")
            local = local.split("+")[0].replace(".", "")
            unique_emails.add(local + "@" + domain)
        return len(unique_emails)


# https://leetcode.com/problems/unique-email-addresses/

File: solutions/_965_unique_email_addresses/solution.java
package _965_unique_email_addresses;

import java.util.*;

class Solution {

  public int numUniqueEmails(String[] emails) {
    Set<String> uniqueEmails = new HashSet<>();
    for (String email : emails) {
      String[] names = email.split("@");
      String local = names[0];
      String domain = names[1];

      local = local.split("\\+")[0].replace(".", "");
      uniqueEmails.add(local + "@" + domain);
    }
    return uniqueEmails.size();
  }
}
/*
https://leetcode.com/problems/unique-email-addresses/
*/
